cdk-addons/apply

```
@@ -39,21 +39,15 @@ def render_templates():
             "num_nodes": node_count,
             "dns_memory_limit": "170Mi"
         },
-        # nanny_memory for heapster
-        # formula from https://github.com/kubernetes/kubernetes/blob/v1.9.6/cluster/addons/cluster-monitoring/influxdb/heapster-controller.yaml#L11
-        "nanny_memory": str(90 * 1024 + node_count * 200) + "Ki",
-        # heapster server information taken from https://github.com/kubernetes/kubernetes/blob/8bd0a306b599bc8511a238c8c69c03c729767350/cluster/gce/gci/configure-helper.sh#L2349
-        "base_metrics_cpu": "80m",
-        "metrics_cpu_per_node": "0.5",
-        "base_metrics_memory": "140Mi",
-        "metrics_memory_per_node": "4",
-        "heapster_min_cluster_size": "16",
 
         # metrics server information taken from https://github.com/kubernetes/kubernetes/blob/8bd0a306b599bc8511a238c8c69c03c729767350/cluster/gce/gci/configure-helper.sh#L2402
         "base_metrics_server_cpu": "40m",
         "base_metrics_server_memory": "40Mi",
         "metrics_server_memory_per_node": "4",
         "metrics_server_min_cluster_size": "16",
+
+        # TODO: with removal of ./addons/cluster-monitoring, this is no longer used.
+        # Decide if we should remove monitorstorage support from here and k8s-master.
```

@Cynerva I believe `monitorstorage` can also be removed, but it will also need to come out of https://github.com/charmed-kubernetes/charm-kubernetes-master/blob/master/config.yaml#L291. I'm not sure about what happens if we remove a snap config option and a user upgrades a < 1.18 snap -- will snapd error if previously valid config is no longer present?

Until I find that out, i'm leaving this helpful `TODO` in case we do end up scrubbing this in 1.18.

---

> I'm not sure about what happens if we remove a snap config option and a user upgrades a < 1.18 snap -- will snapd error if previously valid config is no longer present?

Nah, snapd doesn't care or even know what configs the snap has. Removing the monitorstorage config from cdk-addons 1.18 is safe. It will silently ignore whatever is passed in.

Removing the config from kubernetes-master is trickier, since cdk-addons 1.17 and earlier will error and abort if the monitorstorage config isn't set (because `get_snap_config("monitorstorage")` was not called with `required=False`). So I think the safe thing to do there is keep the monitorstorage config in kubernetes-master, always pass it through to cdk-addons, and update the description to indicate it's deprecated on k8s 1.18+.

---

> So I think the safe thing to do there is keep the monitorstorage config in kubernetes-master, always pass it through to cdk-addons, and update the description to indicate it's deprecated on k8s 1.18+.

Done in https://github.com/charmed-kubernetes/charm-kubernetes-master/pull/73